wrappers/javascript/packages/anoncreds-nodejs/src/NodeJSAnoncreds.ts

```diff
@@ -617,6 +627,380 @@ export class NodeJSAnoncreds implements Anoncreds {
+      presentation,
+      presentationRequest,
+      schemas,
+      credentialDefinitions,
+      revocationRegistryDefinitions,
+      revocationStatusLists,
+      revocationRegistryDefinitionIds,
+      schemaIds,
+      credentialDefinitionIds
+    } = serializeArguments(options)
+
+    const nativeNonRevokedIntervalOverrides = options.nonRevokedIntervalOverrides?.map((value) => {
+      const { requestedFromTimestamp, revocationRegistryDefinitionId, overrideRevocationStatusListTimestamp } =
+        serializeArguments(value)
+      return NonRevokedIntervalOverrideStruct({
+        rev_reg_def_id: revocationRegistryDefinitionId,
+        requested_from_ts: requestedFromTimestamp,
+        override_rev_status_list_ts: overrideRevocationStatusListTimestamp
+      })
+    })
+
+    const nonRevokedIntervalOverrideList = NonRevokedIntervalOverrideListStruct({
+      count: options.nonRevokedIntervalOverrides?.length ?? 0,
+      data: nativeNonRevokedIntervalOverrides as unknown as TypedArray<
+        StructObject<{
+          rev_reg_def_id: string
+          requested_from_ts: number
+          override_rev_status_list_ts: number
+        }>
+      >
+    })
+
+    const ret = allocateInt8Buffer()
+
+    this.nativeAnoncreds.anoncreds_verify_w3c_presentation(
+      presentation,
+      presentationRequest,
+      schemas,
+      schemaIds,
+      credentialDefinitions,
+      credentialDefinitionIds,
+      revocationRegistryDefinitions,
+      revocationRegistryDefinitionIds,
+      revocationStatusLists,
+      nonRevokedIntervalOverrideList as unknown as Buffer,
+      ret
+    )
+    this.handleError()
+
+    return Boolean(handleReturnPointer<number>(ret))
+  }
+
+  public credentialToW3C(options: { objectHandle: ObjectHandle; credentialDefinition: ObjectHandle }): ObjectHandle {
+    const { objectHandle, credentialDefinition } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_credential_to_w3c(objectHandle, credentialDefinition, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public credentialFromW3C(options: { objectHandle: ObjectHandle }): ObjectHandle {
+    const { objectHandle } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_credential_from_w3c(objectHandle, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialAddNonAnoncredsIntegrityProof(options: {
+    objectHandle: ObjectHandle
+    proof: string
+  }): ObjectHandle {
+    const { objectHandle, proof } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_w3c_credential_add_non_anoncreds_integrity_proof(objectHandle, proof, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialAddNonAnonCredsIntegrityProof(options: {
```

Comment:
As mentioned in other PR, I don't think this should be handled by the AnonCreds library